gym/envs/box2d/car_racing.py

```diff
@@ -438,235 +436,209 @@ def step(self, action):
+            )
+        else:
+            return self.isopen
+
+    def render_road(self, zoom, translation, angle):
+        bounds = PLAYFIELD
+        field = [
+            (2 * bounds, 2 * bounds),
+            (2 * bounds, 0),
+            (0, 0),
+            (0, 2 * bounds),
         ]
-
-        k = PLAYFIELD / 20.0
-        colors.extend([0.4, 0.9, 0.4, 1.0] * 4 * 20 * 20)
-        for x in range(-20, 20, 2):
-            for y in range(-20, 20, 2):
-                polygons_.extend(
+        trans_field = []
+        for coord in field:
+            coord = pygame.math.Vector2(coord).rotate_rad(angle)
+            coord = (coord[0] * zoom + translation[0], coord[1] * zoom + translation[1])
+            trans_field.append(coord)
+        gfxdraw.aapolygon(self.surf, trans_field, (102, 204, 102))
+        gfxdraw.filled_polygon(self.surf, trans_field, (102, 204, 102))
+
+        k = bounds / (20.0)
+        grass = []
+        for x in range(0, 40, 2):
+            for y in range(0, 40, 2):
+                grass.append(
                     [
-                        k * x + k,
-                        k * y + 0,
-                        0,
-                        k * x + 0,
-                        k * y + 0,
-                        0,
-                        k * x + 0,
-                        k * y + k,
-                        0,
-                        k * x + k,
-                        k * y + k,
-                        0,
+                        (k * x + k, k * y + 0),
+                        (k * x + 0, k * y + 0),
+                        (k * x + 0, k * y + k),
+                        (k * x + k, k * y + k),
                     ]
                 )
+        for poly in grass:
+            poly = [pygame.math.Vector2(c).rotate_rad(angle) for c in poly]
+            poly = [
+                (c[0] * zoom + translation[0], c[1] * zoom + translation[1])
+                for c in poly
+            ]
+            gfxdraw.aapolygon(self.surf, poly, (102, 230, 102))
+            gfxdraw.filled_polygon(self.surf, poly, (102, 230, 102))
 
         for poly, color in self.road_poly:
-            colors.extend([color[0], color[1], color[2], 1] * len(poly))
-            for p in poly:
-                polygons_.extend([p[0], p[1], 0])
-
-        vl = pyglet.graphics.vertex_list(
-            len(polygons_) // 3, ("v3f", polygons_), ("c4f", colors)
-        )  # gl.GL_QUADS,
-        vl.draw(gl.GL_QUADS)
-        vl.delete()
+            # converting to pixel coordinates
+            poly = [(p[0] + PLAYFIELD, p[1] + PLAYFIELD) for p in poly]
+            poly = [pygame.math.Vector2(c).rotate_rad(angle) for c in poly]
+            poly = [
+                (c[0] * zoom + translation[0], c[1] * zoom + translation[1])
+                for c in poly
+            ]
+            color = [int(c * 255) for c in color]
+            gfxdraw.aapolygon(self.surf, poly, color)
+            gfxdraw.filled_polygon(self.surf, poly, color)
```

Comment:
With the refactoring this would become:
```python
poly = [(p[0] + PLAYFIELD, p[1] + PLAYFIELD) for p in poly]
color = [int(c * 255) for c in color]
draw_colored_polygon(poly, color, zoom, translation)
```